src/Coff/Object.zig

```diff
@@ -145,44 +104,39 @@ fn parseShdrs(self: *Object, allocator: Allocator) !void {
 
 fn parseSymtab(self: *Object, allocator: Allocator) !void {
     const offset = self.header.pointer_to_symbol_table;
+    if (offset == 0) return error.NoSymtab;
     try self.file.seekTo(offset);
 
-    try self.symtab.ensureTotalCapacity(allocator, self.header.number_of_symbols);
+    const size = self.header.number_of_symbols * coff.Symbol.sizeOf();
+    var symtab_buffer = try allocator.alloc(u8, size);
+    errdefer allocator.free(symtab_buffer);
 
-    var i: usize = 0;
-    var num_aux: usize = 0;
-    while (i < self.header.number_of_symbols) : (i += 1) {
-        const symbol = try self.file.reader().readStruct(Symbol);
-
-        // Ignore symbol if it has invalid section number
-        if (symbol.sect_num < 1 or symbol.sect_num > self.shdrtab.items.len) {
-            continue;
-        }
-
-        // Ignore auxillary symbols
-        if (num_aux > 0) {
-            num_aux -= 1;
-            continue;
-        }
-
-        // Check for upcoming auxillary symbols
-        if (symbol.num_aux != 0) {
-            num_aux = symbol.num_aux;
-        }
-
-        self.symtab.appendAssumeCapacity(symbol);
+    const read = try self.file.reader().readAll(symtab_buffer);
+    if (read < size) {
+        log.debug("Expected size: {}, got: {}", .{ size, read });
+        return error.SymtabTooSmall;
```

Comment:
Let's use the error handing mechanism here:
```suggestion
        return coff_file.base.fatal("{s}: expected symtab size {d}, got {d}", .{ size, read });
```